cargo/cargo_build_script_runner/bin.rs

```diff
@@ -53,6 +53,23 @@ fn run_buildrs() -> Result<(), String> {
     create_dir_all(&out_dir_abs)
         .unwrap_or_else(|_| panic!("Failed to make output directory: {:?}", out_dir_abs));
 
+    // Symlink the execroot to the manifest_dir so that we can use relative paths in the arguments.
+    let exec_root_paths = std::fs::read_dir(&exec_root)
```

Comment:
Ok, I think the issue here is that when running outside of a sandbox there will actually be more files in this directory than expected. Instead of reading from the execroot, is there a MANIFEST file or something that tracks all inputs to an action? I forget if that's created for actions or just tests. The idea would be you can link only the things the build script has as an explicitly registered as an input.

Reply:
Looks like MANIFEST is only found in runfiles

Reply:
Thanks @UebelAndre! I can reproduce the issue locally with this diff now:
```diff
diff --git a/cargo/private/cargo_build_script.bzl b/cargo/private/cargo_build_script.bzl
index e567c1c5..3d1f4858 100644
--- a/cargo/private/cargo_build_script.bzl
+++ b/cargo/private/cargo_build_script.bzl
@@ -242,6 +242,9 @@ def _cargo_build_script_impl(ctx):
         mnemonic = "CargoBuildScriptRun",
         progress_message = "Running Cargo build script {}".format(pkg_name),
         env = env,
+        execution_requirements = {
+            "local": "",
+        }
     )
 
     return [
```

What do you think the behavior should be in this case? Should I just disable all of this logic if you're building outside of a bazel sandbox since that's the whole point of this change?

Reply:
My most recent change updates the test to only assert that the execroot is empty if the test is running in a sandbox. This resolves almost all of the issues but now I have 3 remaining. One is definitely symlink related and the other two look unrelated but may be

Questions:
1. Would it be acceptable to make this a feature which does not exist when running on Windows so that it can entirely avoid the windows symlinking issues?
2. If that's not acceptable, would it be acceptable to make this feature opt-in (perhaps through an environment variable) so that consumers of this change can toggle it on where they see fit? That would give folks the ability to enable it for windows in the environments where they have symlinks enabled.

Reply:
I think making this a feature would be good. It can be disabled by default and enabled globally via `--features` in user's `.bazelrc` files. My experience is doing anything on windows is a bad idea 😅 but we should probably aim to keep the behavior consistent across operating systems.